internal/python/plan.go

```diff
@@ -105,7 +112,7 @@ func DeterminePackageManager(ctx *pythonPlanContext) types.PackageManager {
 // HasDependency checks if the specified dependency is in the project.
 func HasDependency(ctx *pythonPlanContext, dependency string) bool {
 	src := ctx.Src
-	pm := DeterminePackageManager(ctx)
+	pm := ctx.PackageManager.Unwrap()
```

Comment:
Better to call `DeterminePackageManager()` instead of using the `ctx.PackageManager` field directly.

The call cost of `DeterminePackageManager()` is low – it basically returns `ctx.PackageManager` if the value has already been set:

https://github.com/zeabur/zbpack/pull/86/files#diff-0bbf727b8ad8b0650c8c7579ead9a0d87b50148ba75224e908cf86d1514a65ddR90

Additionally, having all `PackageManager` requirements go through our `DeterminePackageManager()` simplifies logging and potential pre-determination in the `PackageManager` getter. Think of it as the `PackageManager` getter potentially doing something extra.